{"summary": "JET: 3 pass, 2 failed, 2 error, 1 warning", "tests": [{"type": "Warning", "name": "Warning", "description": "This is an example test that raises a warning", "line": 26, "variables": {}, "out": "", "test": {"name": "Error capturing", "doc": "Error capturing", "module": {"name": "Capture options", "doc": "Examples on how JET deals with local variables and outputs.\n", "path": "/Users/antonio/Documents/Projects/JET_JetTesting/tests/test_capture_options.py", "module": null}, "routine": null}}, {"type": "Failed", "name": "AssertionError", "description": "If test fails, standard output and local variables be viewed in the report", "line": 20, "variables": {"a": ["banana", "book", "apple"], "b": ["banana", "nook", "apple"]}, "out": "this is the standard output of a function\n", "test": {"name": "Failing test with print function", "doc": "Standard ouput is blocked by default", "module": {"name": "Capture options", "doc": "Examples on how JET deals with local variables and outputs.\n", "path": "/Users/antonio/Documents/Projects/JET_JetTesting/tests/test_capture_options.py", "module": null}, "routine": null}}, {"type": "Failed", "name": "AssertionError", "description": "Error message is displayed if test fails", "line": 21, "variables": {}, "out": "", "test": {"name": "Simple test fail", "doc": "Simple test fail", "module": {"name": "Verbose levels", "doc": "Examples of how JET displays test names.\nThese functions should all pass and show what is written when they don't.\n", "path": "/Users/antonio/Documents/Projects/JET_JetTesting/tests/test_verbose_levels.py", "module": null}, "routine": null}}, {"type": "Error", "name": "PatienceError", "description": "CUSTOM ERROR: The function called test_timings_of_calculation exceded my patience", "line": 23, "variables": {"elapsed": 1.0050455419986974, "test_function": "test_timings_of_calculation"}, "out": "", "test": {"name": "Timings of calculation", "doc": "The function should not exeed 0.5 seconds.", "module": {"name": "Custom failing conditions", "doc": "Example of how to add new custom failing conditions.\nThis module shows how to add a custom error condition to your tests.\nFor example a maximum running time.\n", "path": "/Users/antonio/Documents/Projects/JET_JetTesting/tests/test_custom_failing_conditions.py", "module": null}, "routine": null}}, {"type": "Error", "name": "JetError", "description": "Could not run test. Please supply all arguments as default arguments or none at all", "line": 36, "variables": [], "out": "", "test": {"name": "Username", "doc": "Username", "module": {"name": "Custom failing conditions", "doc": "Example of how to add new custom failing conditions.\nThis module shows how to add a custom error condition to your tests.\nFor example a maximum running time.\n", "path": "/Users/antonio/Documents/Projects/JET_JetTesting/tests/test_custom_failing_conditions.py", "module": null}, "routine": null}}]}